### PR TITLE
Feat/add generic connected hook

### PR DESCRIPTION
**Pull request by dpanta94**
Similar to #87 

but here adding:

generic connect hook
specific and generic disconnect hooks
tests :D

### CODE REVIEW

File: src/Uplink/Auth/Admin/Connect_Controller.php [hunk 1]
```diff
@@ -179,5 +179,14 @@ public function maybe_store_token_data(): void {
...
 		do_action( 'stellarwp/uplink/' . Config::get_hook_prefix() . '/' . $slug . '/connected', $plugin );
+
+		/**
+		 * Fires after a plugin has been connected.
+		 *
+		 * @since 2.2.2
+		 *
+		 * @param \StellarWP\Uplink\Resources\Resource $plugin The plugin that was connected.
+		 */
+		do_action( 'stellarwp/uplink/' . Config::get_hook_prefix() . '/connected', $plugin );
```
Comment (redscar) [inline src/Uplink/Auth/Admin/Connect_Controller.php:190]:
```suggestion
		do_action( 'stellarwp/uplink/' . Config::get_hook_prefix() . '/connected', $plugin, $slug );
```

Just a suggestion, would passing `$slug` be helpful?

Comment (dpanta94) [reply src/Uplink/Auth/Admin/Connect_Controller.php:190]: Nope we pass $plugin - slug is just `$plugin->get_slug()` ;)

**Review by redscar — APPROVED** [review]

**Comment by rafsuntaskin** [comment]
@dpanta94 `@since` tags needs to be updated. Other than that looking good 💯

**Review by borkweb — APPROVED** [review]
That's a great addition!

**Pull request merged**